src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shapes = 'triangle' | 'rectangle' | 'circle';
+type Colors = 'red' | 'blue' | 'green';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('This figure does not have a negative value');
```

Comment:
The error message 'This figure does not have a negative value' is misleading. It would be clearer to specify the exact condition, such as 'Side lengths must be positive'.